pkg/providers/amifamily/types.go

```diff
@@ -36,33 +37,53 @@ const (
 )
 
 type AMI struct {
-	Name         string
-	AmiID        string
-	CreationDate string
-	Requirements scheduling.Requirements
+	Name            string
+	AmiID           string
+	CreationDate    string
+	DeprecationTime string
```

Comment:
Seems like we should make this a `bool`, and abstract away the idea of the Deprecation time

Reply:
Addressed